contracts/RelayerHub.sol

```diff
@@ -8,88 +8,193 @@ import "./System.sol";
+    event addManagerByGovEvent(address _addedManager);
+    event registerManagerEvent(address _registeredManager);
+    event editRelayerEvent(address _relayerToBeAdded);
+    event removeRelayerEvent(address _removedRelayer);
+
+
+    function init() external onlyNotInit {
+        requiredDeposit = INIT_REQUIRED_DEPOSIT;
+        dues = INIT_DUES;
+        alreadyInit = true;
+
+        // todo initialise the currently existing Managers and their relayer keys
+
+        managers[WHITELIST_1] = manager(requiredDeposit, dues);
+        managers[WHITELIST_2] = manager(requiredDeposit, dues);
+
+        managersRegistered[WHITELIST_1] = true;
+        managersRegistered[WHITELIST_2] = true;
+
+        relayManagersExistMap[WHITELIST_1] = true;
+        relayManagersExistMap[WHITELIST_2] = true;
+
+        managersAndRelayers[WHITELIST_1] = WHITELIST_1; // fixme current relayer
+        managersAndRelayers[WHITELIST_2] = WHITELIST_2; // fixme current relayer
+
+        // fixme initialise relayerExistsMap
+    }
+
+    /*********************** Param update ********************************/
+    function updateParam(string calldata key, bytes calldata value) external override onlyInit onlyGov {
+        if (Memory.compareStrings(key, "requiredDeposit")) {
+            require(value.length == 32, "length of requiredDeposit mismatch");
+            uint256 newRequiredDeposit = BytesToTypes.bytesToUint256(32, value);
+            require(newRequiredDeposit > 1 && newRequiredDeposit <= 1e21 && newRequiredDeposit > dues, "the requiredDeposit out of range");
+            requiredDeposit = newRequiredDeposit;
+        } else if (Memory.compareStrings(key, "dues")) {
+            require(value.length == 32, "length of dues mismatch");
+            uint256 newDues = BytesToTypes.bytesToUint256(32, value);
+            require(newDues > 0 && newDues < requiredDeposit, "the dues out of range");
+            dues = newDues;
+        } else if (Memory.compareStrings(key, "addManager")) {
+
+            require(value.length == 20, "length of manager address mismatch");
+            address newManager = BytesToTypes.bytesToAddress(20, value);
+            addManagerByGov(newManager);
+
+        } else if (Memory.compareStrings(key, "removeManager")) {
+
+            require(value.length == 20, "length of manager address mismatch");
+            address payable managerAddress = payable(BytesToTypes.bytesToAddress(20, value));
+            removeManagerByGov(managerAddress);
+
+        } else {
+            require(false, "unknown param");
+        }
+        emit paramChange(key, value);
+    }
+
+    function removeManagerByGov(address payable managerToBeRemoved) internal {
+        removeManagerHelper(managerToBeRemoved);
+    }
+
+    function removeManager() external onlyRegisteredManager {
+        // here the manager removes himself
+        removeManagerHelper(msg.sender);
+    }
+
+    function removeManagerHelper(address payable managerAddress) internal {
```

Comment:
I would rename this to `removeManager` and use another name for the function above, something to match its purpose (manager removes himself). A couple of options could be:

- `removeManagerByHimself`
- `removeMe`
- `removeMeFromManagers`